### PR TITLE
Remove Type::Array variant

### PR DESCRIPTION
**Pull request by brendanzab**
Working towards #50

### CODE REVIEW

**Comment by bjz** [comment]
Bump!

**Comment by Marwes** [comment]
For some reason I thought you weren't done since it was so tiny compared to the previous removals!

Looks good, merge if it is indeed done!

**Comment by Marwes** [comment]
Oops, sec noticed a mistake.

File: check/src/kindcheck.rs [hunk 1]
```diff
@@ -139,14 +139,9 @@ impl<'a> KindCheck<'a> {
                 Ok((gen.kind.clone(), Type::generic(gen)))
             }
             Type::Variable(_) => panic!("kindcheck called on variable"),
+            Type::Builtin(BuiltinType::Array) |
```
Comment (Marwes) [inline check/src/kindcheck.rs]: Array and function do not have the same kind (`Type -> Type` vs `Type -> Type -> Type`). If you can add a test to catch this as well that would be great!

Comment (bjz) [inline check/src/kindcheck.rs]: Ah, durp

**Comment by bjz** [comment]
Fixed the array kind issue, but still need to do some tests. Should there be a separate kindcheck test module?

**Comment by Marwes** [comment]
There probably should be a separate kind check module. Currently the tests are just dumped into a single file (did split them into pass and fail in #61 though)

**Comment by bjz** [comment]
arrrg skeptic arrg!

**Comment by Marwes** [comment]
It might be that travis_wait is messing it up, gonna run some builds on travis in #66

**Comment by Marwes** [comment]
@bjz Should this be merged? Or are you working on adding a kindcheck test?

**Comment by brendanzab** [comment]
Yeah - would be nice to merge it - won't have much time to do stuff on gluon this week. Maybe we should make an issue to add a test...

**Comment by Marwes** [comment]
Ok, I think its fine to merge it. Opening an issue for adding kindcheck tests would be a good idea.

**Pull request merged**